wrappers/map.js

```diff
@@ -117,20 +119,42 @@ class Map extends EventEmitter {
 
     this.geojsonLayers[name] = layer
 
+    this.sortLayers()
+
     map.emit('geojson.add', {
       name: name,
       layer: layer
     })
   }
 
+  sortLayers() {
+    var sortableLayers = []
+    Object.keys(this.geojsonLayers).forEach((name) => {
+      var layer = this.geojsonLayers[name]
+      if (!isNaN(parseInt(nn(layer)('options.sortOrder').val, 10))) {
+        sortableLayers.push(layer)
+      }
+    })
+    sortableLayers.sort(function (a, b) {
+      return a.options.sortOrder - b.options.sortOrder
+    })
+    sortableLayers.forEach(function (layer) {
+      try {
+        layer.bringToFront()
+      } catch (e) {}
+    })
```

Comment:
Does it work on an object?
On 10/02/2016 13:00, "Mark Stuart" <notifications@github.com> wrote:

> In wrappers/map.js
> <https://github.com/mediasuitenz/mappy/pull/100#discussion_r52396159>:
>
> > +  sortLayers() {
> > +    var sortableLayers = []
> > +    Object.keys(this.geojsonLayers).forEach((name) => {
> > +      var layer = this.geojsonLayers[name]
> > +      if (!isNaN(parseInt(nn(layer)('options.sortOrder').val, 10))) {
> > +        sortableLayers.push(layer)
> > +      }
> > +    })
> > +    sortableLayers.sort(function (a, b) {
> > +      return a.options.sortOrder - b.options.sortOrder
> > +    })
> > +    sortableLayers.forEach(function (layer) {
> > +      try {
> > +        layer.bringToFront()
> > +      } catch (e) {}
> > +    })
>
> Actually, the map is really a filter
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mediasuitenz/mappy/pull/100/files#r52396159>.
>